src/main/kotlin/blackjack/Bank.kt

```diff
@@ -0,0 +1,116 @@
+package blackjack
+
+class Bank(accounts: List<ParticipantAccount> = emptyList()) {
+    private val accounts: MutableList<ParticipantAccount> = accounts.toMutableList()
+
+    fun bet(
+        participant: Participant,
+        betAmount: Double = 0.0,
+    ) {
+        if (participant is Player) {
+            require(betAmount > 0) {
+                "플레이어는 1원 이상 베팅해야 합니다 betAmount: $betAmount"
+            }
+        }
+        accounts.add(
+            ParticipantAccount(
+                participant = participant,
+                initialBalance = betAmount,
+            ),
+        )
+    }
+
+    fun balance(participant: Participant): Double {
+        return accounts.find { it.participant == participant }?.currentBalance ?: 0.0
+    }
+
+    fun settleBets(gameResults: List<GameResult>) = gameResults.forEach(::settleBet)
+
+    fun settleBet(gameResult: GameResult) {
+        val dealerAccount = findDealerAccount()
+        val playerAccount =
+            findAccount(gameResult.player)
+                ?: throw IllegalStateException("플레이어 계좌를 찾지 못했습니다: player=${gameResult.player}")
+
+        settleOutCome(gameResult.outcome, dealerAccount, playerAccount)
+    }
+
+    fun profits(): List<ParticipantProfit> {
+        return accounts.map {
+            ParticipantProfit(
+                it.participant,
+                it.profit,
+            )
+        }
+    }
+
+    private fun settleOutCome(
+        outcome: Outcome,
+        dealerAccount: ParticipantAccount,
+        playerAccount: ParticipantAccount,
+    ) {
+        when (outcome) {
+            Outcome.WIN -> settleWin(dealerAccount, playerAccount)
+            Outcome.LOSS -> settleLoss(dealerAccount, playerAccount)
+            Outcome.BLACKJACK -> settleBlackJack(dealerAccount, playerAccount)
+            Outcome.PUSH -> Unit
+        }
```

Comment:
각 케이스에 대한 핸들링 로직이 하나의 객체에서 함수로 구분되고 있다보니, 코드가 너무 길어지는 것 같은데, seald class나 책임사슬 패턴을 통해 확장성을 챙겨보면 어떨까요? 
```suggestion
        outcomHandlerFactory.findHandler(outcome).handle(dealerAccount, playerAccount)
```